service/gcs/transport/mock.go

```diff
@@ -15,43 +19,54 @@ type MockTransport struct {
 
 // Dial ignores the port, and returns a MockTransport struct.
 func (t *MockTransport) Dial(_ uint32) (Connection, error) {
-	csr, csw := io.Pipe()
-	scr, scw := io.Pipe()
+	sockDir, err := ioutil.TempDir("", "mocktport")
+	if err != nil {
+		return nil, err
+	}
+	name := filepath.Join(sockDir, "mocktport.sock")
+	addr, err := net.ResolveUnixAddr("unix", name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to resolve unix socket address %s", name)
+	}
+
+	// Set up server connection.
+	listener, err := net.ListenUnix("unix", addr)
```

Comment:
i.e. syscall.Socketpair + net.FileConn

---
In reply to: [132545253](https://github.com/Microsoft/opengcs/pull/106#discussion_r132545253) [](ancestors = 132545253)